src/Orleans.Runtime/Placement/SiloServicePlacementDirector.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Orleans.GrainDirectory;
+using Orleans.Placement;
+
+namespace Orleans.Runtime.Placement
+{
+    internal class SiloServicePlacementDirector : IPlacementDirector, IActivationSelector
+    {
+        private readonly SafeRandom random = new SafeRandom();
+
+        public virtual Task<PlacementResult> OnSelectActivation(
+            PlacementStrategy strategy, GrainId target, IPlacementRuntime runtime)
+        {
+            return TrySelectActivationSynchronously(strategy, target, runtime, out PlacementResult placementResult)
+                ? Task.FromResult(placementResult)
+                : Task.FromResult(default(PlacementResult));
+        }
+
+        public bool TrySelectActivationSynchronously(
+            PlacementStrategy strategy, GrainId target, IPlacementRuntime runtime, out PlacementResult placementResult)
+        {
+            AddressesAndTag addressesAndTag;
+            if (runtime.FastLookup(target, out addressesAndTag) && addressesAndTag.Addresses.Count > 0)
```

Comment:
We can avoid the directory entirely

Reply:
(resolved: this is correct behavior)

Reply:
Can, but it doesn't hurt to call it in case addr is in local cache.  Sticking to normal flow as much as possible.